my_robo:
  joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 50
 
  head_position_controller: 
    type: "position_controllers/JointPositionController"
    joint: 'head_joint'
    pid: {p: 100.0, i: 0.01, d: 10.0}
    publish_rate: 50.0               # default: 50
    
  neck_position_controller: 
    type: "position_controllers/JointPositionController"
    joint: 'neck_joint'
    pid: {p: 100.0, i: 0.01, d: 10.0}
    publish_rate: 50.0               # default: 50
    
  diff_drive_controller:
    type        : "diff_drive_controller/DiffDriveController"
    left_wheel  : 'left_wheel_joint'
    right_wheel : 'right_wheel_joint'
    publish_rate: 50.0               # default: 50
    pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
 
    # Wheel separation and diameter. These are both optional.
    # diff_drive_controller will attempt to read either one or both from the
    # URDF if not specified as a parameter
    # (If controller trying to read from a URDF file, we don't need to define values here?)
    # (but comment out this, navigation does not work.)
    wheel_separation : 0.34
    wheel_radius : 0.085
 
    # Wheel separation and radius multipliers
    wheel_separation_multiplier: 1.0 # default: 1.0
    wheel_radius_multiplier    : 1.0 # default: 1.0
 
    # Velocity commands timeout [s], default 0.5
    cmd_vel_timeout: 1.0
     
    # Base frame_id
    base_frame_id: base_footprint
     
    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 1.0  # m/s
        min_velocity           : -1.0 # m/s
        has_acceleration_limits: true
        max_acceleration       : 0.5  # m/s^2
        min_acceleration       : -0.5 # m/s^2
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 0.5  # rad/s
        min_velocity           : -0.5
        has_acceleration_limits: true
        max_acceleration       : 0.2  # rad/s^2
        min_acceleration       : -0.2
